enclave_manager/avalon_enclave_manager/base_enclave_info.py

```diff
@@ -35,150 +32,72 @@ class BaseEnclaveInfo(ABC):
     data obtained post signup.
     """
 
-    _ias = None
-    _epid_group = None
-    _tcf_enclave_info = None
-    _sig_rl_update_time = None
-
     # -------------------------------------------------------
-    def __init__(self, is_sgx_simulator):
+    def __init__(self, config):
 
         # Initialize the keys that can be used later to
         # register the enclave
-        self._is_sgx_simulator = is_sgx_simulator
+        self._config = config.get("EnclaveModule")
+        self._attestation = None
+        if "attestation_type" in self._config:
+            if self._config["attestation_type"] == "EPID":
+                self._attestation = EpidAttestation(config.get(
+                    "EpidAttestation"))
+            else:
+                raise ValueError('Invalid attestation type {}'.format(
+                    self._config["attestation_type"]))
+        else:
+            raise ValueError('Missing attestation type in config')
```

Comment:
attestation is not mandatory (eg: for SGX simulator). If attestation_type is not present in config can we just log a warning and continue ?